### PR TITLE
Media - Add large and medium image sizes

### PR DESCRIPTION
**Pull request by geriux**
**Related PRs:**

- `Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/5007
- `Gutenberg PR` -> https://github.com/WordPress/gutenberg/pull/42178
- `WordPress FluxC PR` -> https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2456
- `WordPress Utils PR` -> https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/108 

This PR adds support for the `large` and `medium` sizes for images, this data is already available from both the REST API and XMLRPC but it is only stored for self-hosted sites, this PR includes REST API WPCOM sites as well. Having these images allows using a lower size within the editor which improves the performance once images are uploaded.

Now once an image is uploaded from the editor, it will use the `large` size if available, if not it will fall back to the `medium` size and if none are available it will use the `full` size. This change is done to the Gutenberg processors for the Image and Gallery block.

A new `getOptimalFileURL` method was added in `MediaFile` to get the final URL. Due to this, some changes to the test mocks were needed, kudos to @mkevins for the help with that.

For test instructions and screencast please check the [Gutenberg PR description](https://github.com/WordPress/gutenberg/pull/42178).

## Regression Notes
1. Potential unintended areas of impact

It should only affect the Gutenberg editor media uploads.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing:
- Image uploads using the editor and the Image and Gallery block
- Gutenberg processors uploading images and closing the editor

3. What automated tests I added (or what prevented me from doing so)

None, we do manual testing with media uploads from the editor.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by derekblank — APPROVED** [review]
I was able to test and verify the steps following the Testing Instructions on https://github.com/WordPress/gutenberg/pull/42178. LGTM! 🚀

**Comment by geriux** [comment]
Hey @mkevins 👋 I was wondering if you would have some spare time to check this PR since you've worked on media uploads and block processors before. We've already tested it in different scenarios so we are looking for another check for the native changes. Please let me know if you can check it out, if not I'll look for someone else. Thank you!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java [hunk 1]
```diff
@@ -44,7 +46,16 @@ public abstract class BlockProcessor {
...
-        mRemoteUrl = org.wordpress.android.util.StringUtils.notNullStr(Utils.escapeQuotes(mediaFile.getFileURL()));
+
+        String mediaURL;
+        if (!TextUtils.isEmpty(mediaFile.getFileUrlLargeSize())) {
+            mediaURL = mediaFile.getFileUrlLargeSize();
+        } else if (!TextUtils.isEmpty(mediaFile.getFileUrlMediumSize())) {
+            mediaURL = mediaFile.getFileUrlMediumSize();
+        } else {
+            mediaURL = mediaFile.getFileURL();
+        }
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java]: I wonder if we can extract this to a method on the `MediaFile`, e.g. something like `mediaFile.getOptimalFileURL`, since we use the same logic [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16875/files#diff-3d055f27ee69cf53a3055662daab1fc04df6e0eba7deadfd63c3a79df8e96dc5R1346). Wdyt?

File: libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java [hunk 2]
```diff
@@ -1343,7 +1343,16 @@ public void onMediaUploadRetry(String localMediaId, MediaType mediaType) {
...
+        String mediaURL;
+        if (!TextUtils.isEmpty(mediaFile.getFileUrlLargeSize())) {
+            mediaURL = mediaFile.getFileUrlLargeSize();
+        } else if (!TextUtils.isEmpty(mediaFile.getFileUrlMediumSize())) {
+            mediaURL = mediaFile.getFileUrlMediumSize();
+        } else {
+            mediaURL = mediaFile.getFileURL();
+        }
+
+        getGutenbergContainerFragment().mediaFileUploadSucceeded(Integer.valueOf(localMediaId), mediaURL,
```
Comment (mkevins) [inline libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java]: Not sure if linting rules changed, but I'm seeing suggestions in AS to replace `Integer.valueOf` with `Integer.parseInt`.. OTOH, there are other places with the same suggestion that are not part of this PR :man_shrugging: . Please consider this non-blocking - just mentioning it, since I noticed it. :smile:

**Comment by mkevins** [comment]
> Please let me know if you can check it out, if not I'll look for someone else.

Thanks for reaching out @geriux :smile: . It looks good, and seems to safely preserve the correct behavior for files that do not have the other sizes. I left a couple of minor comments about the code. Thanks for the work on this!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java (hunk 1 above)
Comment (geriux) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java]: Great suggestion! I added it in WordPress Utils Android [here](https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/108/commits/1d4fcf5679c857dca845ef7419949db66297aba7), and updated the code in https://github.com/wordpress-mobile/WordPress-Android/pull/16875/commits/1fff96fcd8ab09554ae27f9987ebdc00c62f1b9c

File: libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java (hunk 2 above)
Comment (geriux) [reply libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java]: Thanks for the suggestion! I didn't realize that the code was being highlighted by AS 😅. I updated it in https://github.com/wordpress-mobile/WordPress-Android/pull/16875/commits/1fff96fcd8ab09554ae27f9987ebdc00c62f1b9c#diff-3d055f27ee69cf53a3055662daab1fc04df6e0eba7deadfd63c3a79df8e96dc5R1347-R1348

**Comment by geriux** [comment]
> Thanks for reaching out @geriux 😄 . It looks good, and seems to safely preserve the correct behavior for files that do not have the other sizes. I left a couple of minor comments about the code. Thanks for the work on this!

Thank you so much for reviewing the code @mkevins and for the feedback! I've updated it from your comments, let me know if you can give it another check! Thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java [hunk 3]
```diff
@@ -30,6 +30,7 @@ public abstract class BlockProcessor {
     String mLocalId;
     String mRemoteId;
     String mRemoteUrl;
+    String mRemoteImageUrl;
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java]: Why do we introduce another remote URL here? For blocks other than image (or gallery), `getOptimalFileURL` will return the same string as `getFileURL`. Could we just set `mRemoteURL` to the escaped `mediaFile.getOptimalFileURL`? Wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.java [hunk 4]
```diff
@@ -49,7 +49,7 @@ public GalleryBlockProcessor(String localId, MediaFile mediaFile, String siteUrl
         // if a match is found, proceed with replacement
         if (targetImg != null) {
             // replace attributes
-            targetImg.attr("src", mRemoteUrl);
+            targetImg.attr("src", mRemoteImageUrl);
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.java]: If we make the change from [the suggestion here](https://github.com/wordpress-mobile/WordPress-Android/pull/16875/files#r925143768), we could keep this as `mRemoteUrl` (and the other references as well). Wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java (hunk 1 above)
Comment (mkevins) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java]: > I've updated it from your comments, let me know if you can give it another check!

Thanks for updating it! I've left a couple more minor comments that maybe we could simplify it even further. Thanks for your work on this!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java (hunk 3 above)
Comment (geriux) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java]: I made this change after introducing `getOptimalFileURL`, around 38 tests related to media started failing because of it, so I thought of just changing the needed blocks. Another option is to keep `mRemoteUrl` and update all of the tests to mock `optimalFileUrl`, should we just use `mRemoteUrl` and update the tests?

Comment (mkevins) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java]: > around 38 tests related to media started failing because of it,

:sweat_smile: yikes! thats a lot of failing tests.

> should we just use mRemoteUrl and update the tests?

I still think we should, since it seems weird to have the implementation details of a subclass in the base class. I guess it is natural that we must change the mocks, since we are relying on this new method internally in the object / class under test. But it is rather tedious, so I've made the following patch to "chip in" :smile: :+1: 

<details>
<summary>Patch to update processor tests</summary>

```patch
diff --git a/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java b/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java
index 6f5dc0bc4d..21dba8597c 100644
--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java
@@ -30,7 +30,6 @@ public abstract class BlockProcessor {
     String mLocalId;
     String mRemoteId;
     String mRemoteUrl;
-    String mRemoteImageUrl;
 
     private String mBlockName;
     private JsonObject mJsonAttributes;
@@ -45,9 +44,8 @@ public abstract class BlockProcessor {
     BlockProcessor(String localId, MediaFile mediaFile) {
         mLocalId = localId;
         mRemoteId = mediaFile.getMediaId();
-        mRemoteUrl = org.wordpress.android.util.StringUtils.notNullStr(Utils.escapeQuotes(mediaFile.getFileURL()));
-        mRemoteImageUrl = org.wordpress.android.util.StringUtils.notNullStr(Utils
-                .escapeQuotes(mediaFile.getOptimalFileURL()));
+        mRemoteUrl = org.wordpress.android.util.StringUtils.notNullStr(Utils.escapeQuotes(mediaFile
+                .getOptimalFileURL()));
     }
 
     private JsonObject parseJson(String blockJson) {
diff --git a/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.java b/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.java
index a44ffe8fd3..ebfaea3ced 100644
--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.java
@@ -49,7 +49,7 @@ public class GalleryBlockProcessor extends BlockProcessor {
         // if a match is found, proceed with replacement
         if (targetImg != null) {
             // replace attributes
-            targetImg.attr("src", mRemoteImageUrl);
+            targetImg.attr("src", mRemoteUrl);
             targetImg.attr("data-id", mRemoteId);
             targetImg.attr("data-full-url", mRemoteUrl);
             targetImg.attr("data-link", mAttachmentPageUrl);
@@ -63,7 +63,7 @@ public class GalleryBlockProcessor extends BlockProcessor {
             if (parent != null && parent.is("a") && mLinkTo != null) {
                 switch (mLinkTo) {
                     case "file":
-                        parent.attr("href", mRemoteImageUrl);
+                        parent.attr("href", mRemoteUrl);
                         break;
                     case "post":
                         parent.attr("href", mAttachmentPageUrl);
diff --git a/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/ImageBlockProcessor.java b/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/ImageBlockProcessor.java
index 48864e5c4e..16c26897c6 100644
--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/ImageBlockProcessor.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/ImageBlockProcessor.java
@@ -19,7 +19,7 @@ public class ImageBlockProcessor extends BlockProcessor {
         // if a match is found, proceed with replacement
         if (targetImg != null) {
             // replace attributes
-            targetImg.attr("src", mRemoteImageUrl);
+            targetImg.attr("src", mRemoteUrl);
 
             // replace class
             targetImg.removeClass("wp-image-" + mLocalId);
diff --git a/WordPress/src/test/java/org/wordpress/android/ui/posts/PostUtilsUploadProcessingTest.kt b/WordPress/src/test/java/org/wordpress/android/ui/posts/PostUtilsUploadProcessingTest.kt
index d5346e50dd..7d1086a418 100644
--- a/WordPress/src/test/java/org/wordpress/android/ui/posts/PostUtilsUploadProcessingTest.kt
+++ b/WordPress/src/test/java/org/wordpress/android/ui/posts/PostUtilsUploadProcessingTest.kt
@@ -19,7 +19,7 @@ class PostUtilsUploadProcessingTest {
     @Before
     fun before() {
         whenever(mediaFile.mediaId).thenReturn(TestContent.remoteMediaId)
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteImageUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteImageUrl)
         whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteImageUrl)
         whenever(mediaFile.getAttachmentPageURL(any())).thenReturn(TestContent.attachmentPageUrl)
     }
@@ -49,7 +49,7 @@ class PostUtilsUploadProcessingTest {
 
     @Test
     fun `replaceMediaFileWithUrlInGutenbergPost also works with video`() {
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteVideoUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteVideoUrl)
         val processedContent = PostUtils.replaceMediaFileWithUrlInGutenbergPost(TestContent.oldMediaTextBlockWithVideo,
                 TestContent.localMediaId, mediaFile, TestContent.siteUrl)
         Assertions.assertThat(processedContent).isEqualTo(TestContent.newMediaTextBlockWithVideo)
diff --git a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/AudioBlockProcessorTest.kt b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/AudioBlockProcessorTest.kt
index b6c4aa2322..8f89cbee52 100644
--- a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/AudioBlockProcessorTest.kt
+++ b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/AudioBlockProcessorTest.kt
@@ -14,7 +14,7 @@ class AudioBlockProcessorTest {
     @Before
     fun before() {
         whenever(mediaFile.mediaId).thenReturn(TestContent.remoteMediaId)
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteAudioUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteAudioUrl)
         processor = AudioBlockProcessor(TestContent.localMediaId, mediaFile)
     }
 
diff --git a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/CoverBlockProcessorTest.kt b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/CoverBlockProcessorTest.kt
index 40dce17930..ff5f638012 100644
--- a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/CoverBlockProcessorTest.kt
+++ b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/CoverBlockProcessorTest.kt
@@ -17,7 +17,7 @@ class CoverBlockProcessorTest {
     @Before
     fun before() {
         whenever(mediaFile.mediaId).thenReturn(TestContent.remoteMediaId)
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteImageUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteImageUrl)
         processor = CoverBlockProcessor(TestContent.localMediaId, mediaFile, mediaUploadCompletionProcessor)
     }
 
@@ -43,7 +43,7 @@ class CoverBlockProcessorTest {
     @Test
     fun `processBlock works with outer nested cover blocks`() {
         whenever(mediaFile.mediaId).thenReturn(TestContent.remoteMediaId2)
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteImageUrl2)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteImageUrl2)
         processor = CoverBlockProcessor(TestContent.localMediaId2, mediaFile, mediaUploadCompletionProcessor)
         val processedBlock = processor.processBlock(TestContent.oldCoverBlockWithNestedCoverBlockOuter)
         Assertions.assertThat(processedBlock).isEqualTo(TestContent.newCoverBlockWithNestedCoverBlockOuter)
@@ -63,7 +63,7 @@ class CoverBlockProcessorTest {
 
     @Test
     fun `processBlock works with videos`() {
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteVideoUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteVideoUrl)
         processor = CoverBlockProcessor(TestContent.localMediaId, mediaFile, mediaUploadCompletionProcessor)
         val processedBlock = processor.processBlock(TestContent.oldCoverBlockWithVideo)
         Assertions.assertThat(processedBlock).isEqualTo(TestContent.newCoverBlockWithVideo)
diff --git a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/FileBlockProcessorTest.kt b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/FileBlockProcessorTest.kt
index 20c5f00332..5e217db6be 100644
--- a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/FileBlockProcessorTest.kt
+++ b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/FileBlockProcessorTest.kt
@@ -14,7 +14,7 @@ class FileBlockProcessorTest {
     @Before
     fun before() {
         whenever(mediaFile.mediaId).thenReturn(TestContent.remoteMediaId)
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteImageUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteImageUrl)
         processor = FileBlockProcessor(TestContent.localMediaId, mediaFile)
     }
 
diff --git a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/MediaTextBlockProcessorTest.kt b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/MediaTextBlockProcessorTest.kt
index 06a5c2b1b7..75302ac8d6 100644
--- a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/MediaTextBlockProcessorTest.kt
+++ b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/MediaTextBlockProcessorTest.kt
@@ -15,7 +15,7 @@ class MediaTextBlockProcessorTest {
     @Before
     fun before() {
         whenever(mediaFile.mediaId).thenReturn(TestContent.remoteMediaId)
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteImageUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteImageUrl)
         processor = MediaTextBlockProcessor(TestContent.localMediaId, mediaFile)
     }
 
@@ -33,7 +33,7 @@ class MediaTextBlockProcessorTest {
 
     @Test
     fun `processBlock also works for video`() {
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteVideoUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteVideoUrl)
         processor = MediaTextBlockProcessor(TestContent.localMediaId, mediaFile)
         val processedBlock = processor.processBlock(TestContent.oldMediaTextBlockWithVideo)
         Assertions.assertThat(processedBlock).isEqualTo(TestContent.newMediaTextBlockWithVideo)
diff --git a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/MediaUploadCompletionProcessorTest.kt b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/MediaUploadCompletionProcessorTest.kt
index 09c42f918b..d68903b8fb 100644
--- a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/MediaUploadCompletionProcessorTest.kt
+++ b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/MediaUploadCompletionProcessorTest.kt
@@ -19,7 +19,7 @@ class MediaUploadCompletionProcessorTest {
     @Before
     fun before() {
         whenever(mediaFile.mediaId).thenReturn(TestContent.remoteMediaId)
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteImageUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteImageUrl)
         whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteImageUrl)
         whenever(mediaFile.getAttachmentPageURL(any())).thenReturn(TestContent.attachmentPageUrl)
         processor = MediaUploadCompletionProcessor(TestContent.localMediaId, mediaFile, TestContent.siteUrl)
@@ -33,7 +33,7 @@ class MediaUploadCompletionProcessorTest {
 
     @Test
     fun `processPost splices id and url for a video block`() {
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteVideoUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteVideoUrl)
         processor = MediaUploadCompletionProcessor(TestContent.localMediaId, mediaFile, TestContent.siteUrl)
         val blocks = processor.processContent(TestContent.oldPostVideo)
         Assertions.assertThat(blocks).isEqualTo(TestContent.newPostVideo)
@@ -66,7 +66,7 @@ class MediaUploadCompletionProcessorTest {
     @Test
     fun `processPost works for nested outer cover blocks`() {
         whenever(mediaFile.mediaId).thenReturn(TestContent.remoteMediaId2)
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteImageUrl2)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteImageUrl2)
         processor = MediaUploadCompletionProcessor(TestContent.localMediaId2, mediaFile, TestContent.siteUrl)
         val blocks = processor.processContent(TestContent.oldCoverBlockWithNestedCoverBlockOuter)
         Assertions.assertThat(blocks).isEqualTo(TestContent.newCoverBlockWithNestedCoverBlockOuter)
diff --git a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/VideoBlockProcessorTest.kt b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/VideoBlockProcessorTest.kt
index b548503a5b..355df30e3c 100644
--- a/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/VideoBlockProcessorTest.kt
+++ b/WordPress/src/test/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/VideoBlockProcessorTest.kt
@@ -15,7 +15,7 @@ class VideoBlockProcessorTest {
     @Before
     fun before() {
         whenever(mediaFile.mediaId).thenReturn(TestContent.remoteMediaId)
-        whenever(mediaFile.fileURL).thenReturn(TestContent.remoteVideoUrl)
+        whenever(mediaFile.optimalFileURL).thenReturn(TestContent.remoteVideoUrl)
         processor = VideoBlockProcessor(TestContent.localMediaId, mediaFile)
     } 
```
</details>

We only "save" two lines of code with all of this, but I think it's also _cleaner_ to not "pollute" the base class this way. Wdyt?

Comment (geriux) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java]: Thank you so much for providing the patch! I've applied it in https://github.com/wordpress-mobile/WordPress-Android/pull/16875/commits/12ef532223a7ceb3dfe40747bdb48565ccbc329a

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.java (hunk 4 above)
Comment (geriux) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.java]: Sounds good! I've applied the changes to remove `mRemoteImageUrl`

**Comment by geriux** [comment]
Thanks again for the help with the review @mkevins! After discussing internally and applying the latest changes I'll go ahead and start merging. 🎉
